### PR TITLE
Staging (New User's permission schema and other unrelated changes)

### PR DESCRIPTION
**Pull request by uni2grow**
merge branch feature/main/21 - feature/main/30 without branch feature/main/23

### CODE REVIEW

File: src/main/java/org/isf/permissions/manager/PermissionManager.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package org.isf.permissions.manager;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/permissions/manager/PermissionManager.java:1]: Missing license header

File: pom.xml [hunk 2]
```diff
@@ -307,7 +307,7 @@
 		<dependency>
 			<groupId>org.igniterealtime.smack</groupId>
 			<artifactId>smackx</artifactId>
-			<version>3.4.1</version>
+			<version>3.2.1</version>
```
Comment (dbmalkovsky) [inline pom.xml:310]: why is this dependency down graded?

**Comment by dbmalkovsky** [comment]
This PR appears not to be based on the lasted `develop` branch; please verify and correct as there are conflicts.  Thanks.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
+# Warning
+|:warning: This fork is for _prototyping purpose only_. If you want to submit a PR to the Open Hospital project starts from the [official ISF (Informatici Senza Frontiere) repos](https://github.com/informatici).      |
+|:---------------------------|
+
 # Open Hospital - Core
-[![Java CI](https://github.com/informatici/openhospital-core/workflows/Java%20CI%20with%20Maven/badge.svg)](https://github.com/informatici/openhospital-core/actions?query=workflow%3A%22Java+CI+with+Maven%22)
+[![Java CI](https://github.com/intesys/openhospital-core/workflows/Java%20CI%20with%20Maven/badge.svg)](https://github.com/intesys/openhospital-core/actions?query=workflow%3A%22Java+CI+with+Maven%22)
```
Comment (mwithi) [inline README.md:6]: I think this change should not be included in the PR

File: pom.xml [hunk 4]
```diff
@@ -13,7 +13,7 @@
 	<groupId>org.isf</groupId>
 	<artifactId>OH-core</artifactId>
 	<packaging>jar</packaging>
-	<version>1.11.0</version>
+	<version>2.0.0</version>
```
Comment (mwithi) [inline pom.xml:16]: This change suggests we should redirect this PR into a new branch?

File: src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java [hunk 5]
```diff
@@ -184,7 +184,7 @@ public int newAdmissionReturnKey(Admission admission) throws OHServiceException
 	 * @throws OHServiceException
 	 */
 	public boolean updateAdmission(Admission admission) throws OHServiceException {
-		validateAdmission(admission, false);
+		//validateAdmission(admission, false);
```
Comment (mwithi) [inline src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java:187]: why is validation skipped in this case?

File: rsc/database.properties [hunk 6]
```diff
@@ -0,0 +1,3 @@
+jdbc.url=jdbc:mysql://localhost:3306/oh
+jdbc.username=isf
+jdbc.password=isf123
```
Comment (mwithi) [inline rsc/database.properties:3]: Since v1.11.0 `.properties` files are not included in the repo, only `.properties.dist`

File: src/main/java/org/isf/disease/service/DiseaseIoOperationRepository.java [hunk 7]
```diff
@@ -31,7 +31,8 @@
 
 @Repository
 public interface DiseaseIoOperationRepository extends JpaRepository<Disease, String> {
-
+	
+	@Query(value = "select d FROM Disease d WHERE  d.code = :code")    // d.diseaseType.code = :code
```
Comment (mwithi) [inline src/main/java/org/isf/disease/service/DiseaseIoOperationRepository.java:35]: is this annotation really needed since it's performing the same query as per [JpaRepository query keywords](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repository-query-keywords)?

File: src/main/java/org/isf/opd/model/Opd.java [hunk 8]
```diff
@@ -139,6 +139,21 @@
...
+	@Column(name="OPD_ALLERGIES")
+    private String allergies; // ADDED: Arnaud
+    
+	@Column(name="OPD_THERAPIES")
+    private String therapies; // ADDED: Arnaud
+    
+	@Column(name="OPD_PRESCRIPTION")
+    private String prescription; // ADDED: Arnaud
+	
 	@Transient
```
Comment (mwithi) [inline src/main/java/org/isf/opd/model/Opd.java:157]: it seems this PR is introducing too many feature changes, I think we should split it into different PR and link them to a Jira issue

File: src/main/java/org/isf/operation/manager/OperationBrowserManager.java [hunk 9]
```diff
@@ -28,11 +28,13 @@
 
 import org.isf.generaldata.MessageBundle;
 import org.isf.operation.model.Operation;
+import org.isf.operation.model.OperationRow;
 import org.isf.operation.service.OperationIoOperations;
 import org.isf.opetype.model.OperationType;
 import org.isf.utils.exception.OHServiceException;
 import org.isf.utils.validator.DefaultSorter;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.Query;
```
Comment (mwithi) [inline src/main/java/org/isf/operation/manager/OperationBrowserManager.java:37]: are these import really used?

File: src/main/java/org/isf/operation/service/OperationIoOperationRepository.java [hunk 10]
```diff
@@ -24,6 +24,7 @@
 import org.isf.operation.model.Operation;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
```
Comment (mwithi) [inline src/main/java/org/isf/operation/service/OperationIoOperationRepository.java:27]: Is this import really used?

File: src/main/java/org/isf/operation/service/OperationIoOperations.java [hunk 11]
```diff
@@ -25,10 +25,13 @@
...
 
 import org.isf.operation.model.Operation;
+import org.isf.operation.model.OperationRow;
 import org.isf.opetype.model.OperationType;
 import org.isf.utils.db.TranslateOHServiceException;
 import org.isf.utils.exception.OHServiceException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Service;
```
Comment (mwithi) [inline src/main/java/org/isf/operation/service/OperationIoOperations.java:35]: Are these imports really needed?

File: src/main/java/org/isf/operation/service/OperationRowIoOperationRepository.java [hunk 12]
```diff
@@ -27,6 +27,7 @@
 import org.isf.opd.model.Opd;
 import org.isf.operation.model.OperationRow;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
```
Comment (mwithi) [inline src/main/java/org/isf/operation/service/OperationRowIoOperationRepository.java:30]: Is this import really needed?

File: src/main/java/org/isf/operation/service/OperationRowIoOperations.java [hunk 13]
```diff
@@ -30,6 +30,7 @@
 import org.isf.utils.db.TranslateOHServiceException;
 import org.isf.utils.exception.OHServiceException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.Query;
```
Comment (mwithi) [inline src/main/java/org/isf/operation/service/OperationRowIoOperations.java:33]: Is this import really needed?

**Review by mwithi — CHANGES_REQUESTED** [review]
Very good improvements!

However, I think we should split this PR into two different tasks and link each of them to a Jira issue.

Moreover, since it prepeares for a future release (2.0.0) that may follow other earlier release (1.12.0) we could welcome these changes into a separate branch, wdyt?

Please also check that after https://github.com/informatici/openhospital-core/pull/623 there are some conflicts now.
